extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -324,21 +331,45 @@ export class AmpStoryPage extends AMP.BaseElement {
 
   /**
    * Gets all media elements on this page.
-   * @return {!NodeList<!Element>}
+   * @return {!Array<?Element>}
    * @private
    */
   getAllMedia_() {
-    return this.element.querySelectorAll('audio, video');
+    return this.getMediaBySelector_(Selectors.ALL_MEDIA);
   }
 
 
   /**
    * Gets all video elements on this page.
-   * @return {!NodeList<!Element>}
+   * @return {!Array<?Element>}
    * @private
    */
   getAllVideos_() {
-    return this.element.querySelectorAll('video');
+    return this.getMediaBySelector_(Selectors.ALL_VIDEO);
+  }
+
+
+  /**
+   * Gets media on page by given selector. Finds elements through friendly
+   * iframe (if one exists).
+   * @param {string} selector
+   * @return {!Array<?Element>}
+   */
+  getMediaBySelector_(selector) {
+    const iframe = this.element.querySelector('iframe');
+    const iframeDoc = iframe && iframe.contentDocument;
```

Comment:
did you check if this line will generate console error in case of xorigin iframe?

Reply:
yes it returns `null`